cmd/api/v1/cluster.go

```diff
@@ -573,6 +588,54 @@ func (c *cluster) kubeconfig(args []string) error {
 	return nil
 }
 
+func (c *cluster) execConfig(args []string) error {
+	ctx, cancel := c.c.NewRequestContext()
+	defer cancel()
+
+	id, err := genericcli.GetExactlyOneArg(args)
+	if err != nil {
+		return err
+	}
+
+	ec, err := kubernetes.NewUserExecCache(c.c.Fs)
+	if err != nil {
+		return err
+	}
+
+	creds, err := ec.LoadCachedCredentials(id)
+	if err != nil {
+		// we cannot load cache, so cleanup the cache
+		_ = ec.Clean(id)
+	}
+	if creds == nil {
+		req := &apiv1.ClusterServiceGetCredentialsRequest{
+			Uuid:       id,
+			Project:    c.c.GetProject(),
+			Expiration: durationpb.New(viper.GetDuration("expiration")),
+		}
+
+		resp, err := c.c.Client.Apiv1().Cluster().GetCredentials(ctx, connect.NewRequest(req))
+		if err != nil {
+			return fmt.Errorf("failed to get cluster credentials: %w", err)
+		}
+
+		// TODO: do we really need an expiration from the user?
```

Comment:
Probably the server would need to return the expiration if we do not specify it here? In this case I think it's okay if the client decides the expiration. From the server-side we just reduce the maximum expiration time to 14 days or something.